### ISSUE TITLE
kuma-sidecar returning failed merge error message

### ISSUE THREAD

**Issue by udleinati** [kumahq/kuma#2697]
### Summary

I'm getting constant error messages from kuma-sidecar from all pods, no exceptions.

### Steps To Reproduce

It's a very basic installation, reproduced in 2 different EKS Cluster.

1. Mesh File:

```
apiVersion: kuma.io/v1alpha1
kind: Mesh
metadata:
  name: default
spec:
  logging:
    defaultBackend: fluentd
    backends:
      - name: fluentd
        format: '{"start_time": "%START_TIME%", "source": "%KUMA_SOURCE_SERVICE%", "destination": "%KUMA_DESTINATION_SERVICE%", "source_address": "%KUMA_SOURCE_ADDRESS_WITHOUT_PORT%", "destination_address": "%UPSTREAM_HOST%", "duration_millis": "%DURATION%", "bytes_received": "%BYTES_RECEIVED%", "bytes_sent": "%BYTES_SENT%"}'
        type: tcp
        conf:
          address: 'fluentd.kube-system.svc:9880'
  metrics:
    enabledBackend: prometheus
    backends:
      - name: prometheus
        type: prometheus
        skipMTLS: true
        conf:
          port: 5670
          path: /metrics

---

apiVersion: kuma.io/v1alpha1
kind: TrafficLog
metadata:
  name: all-traffic-log
mesh: default
spec:
  # This TrafficLog policy applies all traffic in that Mesh.
  sources:
    - match:
        kuma.io/service: '*'
  destinations:
    - match:
        kuma.io/service: '*'
  conf:
    backend: fluentd

---

apiVersion: kuma.io/v1alpha1
kind: Retry
mesh: default
metadata:
  name: retry-all-default
spec:
  sources:
    - match:
        kuma.io/service: '*'
  destinations:
    - match:
        kuma.io/service: '*'
  conf:
    http:
      numRetries: 3
      perTryTimeout: 8s
      backOff:
        baseInterval: 0.100s
        maxInterval: 0.250s
      retriableStatusCodes:
        - 503
```

2. Installing from kuma helm:

```
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: kuma
spec:
  releaseName: kuma
  chart:
    spec:
      chart: kuma
      sourceRef:
        kind: HelmRepository
        name: kuma
        namespace: flux-system
      version: "0.6.3"
  interval: 30s
  install:
    remediation:
      retries: 3
  valuesFrom:
    - kind: ConfigMap
      name: kuma-values
```

3. kuma helm config file

```
apiVersion: v1

# https://github.com/kumahq/kuma/tree/master/deployments/charts/kuma
controlPlane:
  logLevel: info
  replicas: 1
```

4. rollout some deployments to start kuma-sidecar

### Additional Details & Logs

- Version

```
EKS Cluster running Kubernetes 1.21.2
Kuma Helm 0.6.3
Kuma 1.2.3
```

- Error logs

```
2021-08-27T09:53:03.614Z	ERROR	metrics-hijacker	unable to merge labels, falling back to unmerged state	{"error": "failed to merge label 'envoy_response_code', values are not equal: '200' != '201'", "errorVerbose": "failed to merge label 'envoy_response_code', values are not equal: '200' != '201'\ngithub.com/kumahq/kuma/app/kuma-dp/pkg/dataplane/metrics.mergeLabels\n\t/home/circleci/project/app/kuma-dp/pkg/dataplane/metrics/merge.go:103\ngithub.com/kumahq/kuma/app/kuma-dp/pkg/dataplane/metrics.merge\n\t/home/circleci/project/app/kuma-dp/pkg/dataplane/metrics/merge.go:63\ngithub.com/kumahq/kuma/app/kuma-dp/pkg/dataplane/metrics.MergeClusters\n\t/home/circleci/project/app/kuma-dp/pkg/dataplane/metrics/merge.go:39\ngithub.com/kumahq/kuma/app/kuma-dp/pkg/dataplane/metrics.(*Hijacker).ServeHTTP\n\t/home/circleci/project/app/kuma-dp/pkg/dataplane/metrics/server.go:109\nnet/http.serverHandler.ServeHTTP\n\t/home/circleci/go/src/net/http/server.go:2887\nnet/http.(*conn).serve\n\t/home/circleci/go/src/net/http/server.go:1952\nruntime.goexit\n\t/home/circleci/go/src/runtime/asm_amd64.s:1371"}
2021-08-27T09:53:03.614Z	ERROR	metrics-hijacker	unable to merge labels, falling back to unmerged state	{"error": "failed to merge label 'envoy_response_code', values are not equal: '200' != '503'", "errorVerbose": "failed to merge label 'envoy_response_code', values are not equal: '200' != '503'\ngithub.com/kumahq/kuma/app/kuma-dp/pkg/dataplane/metrics.mergeLabels\n\t/home/circleci/project/app/kuma-dp/pkg/dataplane/metrics/merge.go:103\ngithub.com/kumahq/kuma/app/kuma-dp/pkg/dataplane/metrics.merge\n\t/home/circleci/project/app/kuma-dp/pkg/dataplane/metrics/merge.go:63\ngithub.com/kumahq/kuma/app/kuma-dp/pkg/dataplane/metrics.MergeClusters\n\t/home/circleci/project/app/kuma-dp/pkg/dataplane/metrics/merge.go:39\ngithub.com/kumahq/kuma/app/kuma-dp/pkg/dataplane/metrics.(*Hijacker).ServeHTTP\n\t/home/circleci/project/app/kuma-dp/pkg/dataplane/metrics/server.go:109\nnet/http.serverHandler.ServeHTTP\n\t/home/circleci/go/src/net/http/server.go:2887\nnet/http.(*conn).serve\n\t/home/circleci/go/src/net/http/server.go:1952\nruntime.goexit\n\t/home/circleci/go/src/runtime/asm_amd64.s:1371"}
2021-08-27T09:53:03.615Z	ERROR	metrics-hijacker	unable to merge labels, falling back to unmerged state	{"error": "failed to merge label 'envoy_response_code_class', values are not equal: '2' != '5'", "errorVerbose": "failed to merge label 'envoy_response_code_class', values are not equal: '2' != '5'\ngithub.com/kumahq/kuma/app/kuma-dp/pkg/dataplane/metrics.mergeLabels\n\t/home/circleci/project/app/kuma-dp/pkg/dataplane/metrics/merge.go:103\ngithub.com/kumahq/kuma/app/kuma-dp/pkg/dataplane/metrics.merge\n\t/home/circleci/project/app/kuma-dp/pkg/dataplane/metrics/merge.go:63\ngithub.com/kumahq/kuma/app/kuma-dp/pkg/dataplane/metrics.MergeClusters\n\t/home/circleci/project/app/kuma-dp/pkg/dataplane/metrics/merge.go:39\ngithub.com/kumahq/kuma/app/kuma-dp/pkg/dataplane/metrics.(*Hijacker).ServeHTTP\n\t/home/circleci/project/app/kuma-dp/pkg/dataplane/metrics/server.go:109\nnet/http.serverHandler.ServeHTTP\n\t/home/circleci/go/src/net/http/server.go:2887\nnet/http.(*conn).serve\n\t/home/circleci/go/src/net/http/server.go:1952\nruntime.goexit\n\t/home/circleci/go/src/runtime/asm_amd64.s:1371"}
```

- Configuration

Get the configuration files from the steps to reproduce it.

- Platform and Operating System

EKS

- Installation Method (Helm, kumactl, AWS CloudFormation, etc.)

Helm using flux-system

**Comment by jpeach** [comment]
Hi @udleinati!

This looks a lot like the bug that was fixed in https://github.com/kumahq/kuma/pull/2476. We just released Kuma 1.3.0, which includes this fix.

**Comment by udleinati** [comment]
Hey @jpeach. It worked. I previously tested it but it was returning something like `go panic` or so, but now in exactly the same instance, I didn't get any issue. Thank you.
